_opensearch/java-rest-high-level.md

```diff
@@ -0,0 +1,111 @@
+---
+layout: default
+title: Java high-level REST client
+nav_order: 97
+---
+
+# Java high-level REST client
+
+The Elasticsearch OSS Java high-level REST client allows you to interact with your OpenSearch clusters and indices through request objects and response objects. To create an index, add data to a document, or complete some other operation, you submit request objects to your cluster, and you get back response objects that have all of the available information, such as the associated index or ID, returned from your cluster.
+
+## Setup
+
+To start using the Elasticsearch OSS Java high-level REST client, ensure that you have the following dependency in your project's `pom.xml` file:
+
+```
+<dependency>
+  <groupId>org.elasticsearch.client</groupId>
+  <artifactId>elasticsearch-rest-high-level-client</artifactId>
+  <version>7.10.2</version>
+</dependency>
+```
+
+You can now start your OpenSearch cluster. Any version of OpenSearch works with the Elasticsearch OSS high-level REST client.
+
+## Create high-level client
+
+Before you begin submitting request objects, you need to first create a high-level REST client object. The following example demonstrates how to establish a client that uses basic authorization credentials to access a cluster.
+
+If you're using the Elasticsearch OSS high-level REST client over HTTPS, you also need to point your java project to a keystore that has the appropriate security certificates.
+
+```java
+System.setProperty("javax.net.ssl.trustStore", "/full/path/to/keystore.jks");
+System.setProperty("javax.net.ssl.trustStorePassword", password-to-keystore);
+
+final CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+
+credentialsProvider.setCredentials(AuthScope.ANY,
+        new UsernamePasswordCredentials("admin", "admin"));
+
+RestClientBuilder builder = RestClient.builder(new HttpHost("localhost", 9200, "https"))
+        .setHttpClientConfigCallback(new RestClientBuilder.HttpClientConfigCallback() {
+            @Override
+            public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpClientBuilder) {
+                return httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider);
+            }
+        });
+
+RestHighLevelClient client = new RestHighLevelClient(builder);
+```
+
+After creating your client, you're ready to start sending requests to your cluster.
+
+## Create index and index documents
+
+The following example demonstrates how to create an index with non-default settings.
+
+```java
+CreateIndexRequest createIndexRequest = new CreateIndexRequest("custom-index");
+
+createIndexRequest.settings(Settings.builder() //Specify in the settings how many shards you want in the index.
+      .put("index.number_of_shards", 4)
+      .put("index.number_of_replicas", 3)
+  );
+
+createIndexRequest.mapping( //Specify any mappings you want.
```

Comment:
Can you not provide a map here? Part of the appeal is that you shouldn't have to be providing strings. The client should handle moving objects to JSON and back again. Is there no option for something like this?

```java
// Create the document as a hash map
Map<String, Object> document = new HashMap<>();
document.put("title", "Walk the Line");
document.put("director", "James Mangold");
document.put("year", "2005");

// Form the indexing request, send it, and print the response
IndexRequest request = new IndexRequest(index, type, id).source(document);
```